From 75274cf4bfea2dad55038ddb50508bd0410838ef Mon Sep 17 00:00:00 2001
From: Liu Ying <Ying.Liu@freescale.com>
Date: Tue, 18 Feb 2014 13:19:12 +0800
Subject: [PATCH 0718/1074] ENGR00298052-2 Documentation: video: add Hannstar
 CABC dt bindings

This patch documents the Hannstar CABC driver's device tree bindings.

Signed-off-by: Liu Ying <Ying.Liu@freescale.com>
[Original patch taken from git://git.freescale.com/imx/linux-2.6-imx.git]
Signed-off-by: Biyao Zhai <biyao.zhai@windriver.com>
---
 .../devicetree/bindings/video/hannstar,cabc.txt    |   28 ++++++++++++++++++++
 1 files changed, 28 insertions(+), 0 deletions(-)
 create mode 100644 Documentation/devicetree/bindings/video/hannstar,cabc.txt

diff --git a/Documentation/devicetree/bindings/video/hannstar,cabc.txt b/Documentation/devicetree/bindings/video/hannstar,cabc.txt
new file mode 100644
index 0000000..bf582ef
--- /dev/null
+++ b/Documentation/devicetree/bindings/video/hannstar,cabc.txt
@@ -0,0 +1,28 @@
+* Hannstar CABC device tree bindings
+
+The Hannstar CABC function may turn a display plane's backlight
+automatically according to the content shown on the plane.  The
+function is controlled(enabled/disabled) by a GPIO.
+
+Required properties:
+- compatible : Should be "hannstar,cabc".
+
+Each standalone CABC gpio is represented as a sub-node of "hannstar_cabc":
+Required subnode properties:
+- gpios: OF device-tree gpio specification.
+
+Optional subnode properties:
+- cabc-enable: Provide this bool property if you want to enable the CABC feature.
+
+Example:
+	hannstar_cabc {
+		compatible = "hannstar,cabc";
+
+		lvds0 {
+			gpios = <&gpio6 15 GPIO_ACTIVE_HIGH>;
+		};
+
+		lvds1 {
+			gpios = <&gpio6 16 GPIO_ACTIVE_HIGH>;
+		};
+	};
-- 
1.7.5.4

